services/fluxmonitor.go

```diff
@@ -366,12 +366,13 @@ func (fm *FluxMonitor) getAdapterResponse(endpoint url.URL, from string, to stri
 }
 
 func (fm *FluxMonitor) poll() error {
-
+	fm.pollMutex.Lock()
```

Comment:
It's a bit safer to add the defer unlock here, so we're sure that it's unlocked if the function returns early from an error.

```suggestion
	fm.pollMutex.Lock()
	defer fm.pollMutex.Unlock()
```